File: leetcode/3100. 换水问题 II.go
func maxBottlesDrunk(numBottles int, numExchange int) int {
	res := 0
	for numBottles > 0 {
		if numBottles >= numExchange {
			res += numExchange
			numBottles -= numExchange - 1
			numExchange++
		} else {
			res += numBottles
			break
		}
	}
	return res
}

File: leetcode/3100. 换水问题 II.rs
impl Solution {
    pub fn max_bottles_drunk(num_bottles: i32, num_exchange: i32) -> i32 {
        let mut res = 0;
        let mut num_bottles = num_bottles;
        let mut num_exchange = num_exchange;
        while num_bottles > 0 {
            if num_bottles >= num_exchange {
                res += num_exchange;
                num_bottles -= num_exchange - 1;
                num_exchange += 1;
            } else {
                res += num_bottles;
                break;
            }
        }
        res
    }
}